backend/packages/Upgrade/src/api/controllers/SegmentController.ts

```diff
@@ -497,6 +498,43 @@ export class SegmentController {
     return this.segmentService.validateSegments(segments, request.logger);
   }
 
+  /**
+   * @swagger
+   * /segments/import/validation:
+   *    post:
+   *       description: Validating Segments with response for Common Import Modal
+   *       consumes:
+   *         - application/json
+   *       parameters:
+   *         - in: body
+   *           name: params
+   *           description: Segment file
+   *           required: true
+   *           schema:
+   *            type: object
```

Comment:
The schema should be updated as it is updated to the array. There are other places where this update needs to be done, which should have been done as part of earlier PRs.
```
type: array
items:
  type: object
  properties:
    fileName:
      type: string
    fileContent:
      type: string
```